### PR TITLE
chore: defers tagging to post-merge

### PR DESCRIPTION
**Pull request by NQuinn27**
This PR defers the tagging until after the release PR has been merged.

Changes in `create-release.yml`
- Use inputs to select which commitizen release type to use
- Use commitizen directly to bump the version without creating a tag
- Create the PR for the release

This PR adds `post-merge.yml`
- In the event a PR has been merged, and the head is `release/next` and the base is `master`, the workflow is run
- The workflow creates a tag based on the current commitizen version

Here is a sample PR for the pre-merge flow: https://github.com/primer-io/primer-sdk-ios/pull/690

### CODE REVIEW

File: .github/workflows/post_merge.yml
Comment (semirp) [inline .github/workflows/post_merge.yml:1]: [nitpick]: Maybe we can rename this to `post-release-merge`?

File: .github/workflows/post_merge.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Install cz
+        run: pip install --user -U Commitizen
+      - name: Record release version
+        run: |
+          echo "RELEASE_VERSION=$(cz version --project)" >> $GITHUB_ENV
+      - name: Tag release
+        uses: tvdias/github-tagger@v0.0.1
```
Comment (semirp) [inline .github/workflows/post_merge.yml:25]: [nitpick]: Maybe we should consider using some more maintained action?

**Review by semirp — APPROVED** [review]

File: .github/workflows/post_merge.yml (hunk 1 above)
Comment (NQuinn27) [reply .github/workflows/post_merge.yml]: I changed this to manually tag. The local git identity should be set correctly having checked out the repo using the RELEASE_ACCESS_TOKEN

**Review by semirp — APPROVED** [review]
LGTM!

**Review by borisprimer — APPROVED** [review]

**Comment by NQuinn27** [comment]
ADR note: https://www.notion.so/primerio/ADR-Release-workflow-defer-tagging-to-post-merge-3582d49acd5b4f9982fd54c81e083e42?pvs=4
Flow diagram for entire flow:
image

**Review by xevious78 — APPROVED** [review]

**Review by xevious78 — APPROVED** [review]

**Pull request merged**